### PR TITLE
Format StmtAssign, badly

### PR DESCRIPTION
**Pull request by konstin**
This is a near-verbatim formatting of assignment statements so we can see the effect of formatting the right hand side. It lacks the wrapping rules for the left hand side and will put the entire left hand side on one line.

On the snapshots this has mostly the effect of normalizing the whitespace around equals sign, but more importantly it will surface all changes to expressions.

### CODE REVIEW

File: crates/ruff_python_formatter/src/statement/stmt_assign.rs [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
-        write!(f, [verbatim_text(item.range)])
+        let StmtAssign {
+            range: _,
+            targets,
+            value,
+            type_comment: _,
+        } = item;
+        write!(
+            f,
+            [LhsAssignList::new(targets), text(" = "), value.format()]
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/statement/stmt_assign.rs:22]: Should this use value.format().with_options(Parenthesize::IfBreaks) to only add parentheses if necessary? Or what's black's behavior?

Comment (konstin) [reply crates/ruff_python_formatter/src/statement/stmt_assign.rs:22]: I couldn't find a symbol like `Parenthesize::IfBreaks`, i now added a group and if_group_breaks which doesn't do the same as black but still at least keeps the line length constraint

Comment (MichaReiser) [reply crates/ruff_python_formatter/src/statement/stmt_assign.rs]: Yeah, this is in the binary expression formatting

Comment (konstin) [reply crates/ruff_python_formatter/src/statement/stmt_assign.rs]: i realized i can change both PRs significantly with the new utils

**Comment by konstin** [comment]
tests are passing locally -.-

**Comment by MichaReiser** [comment]
> tests are passing locally -.-

Can you try a cargo clean and then re-run the tests. It could be that cargo isn't picking up the new file.

File: crates/ruff_python_formatter/src/statement/stmt_assign.rs [hunk 2]
```diff
@@ -1,12 +1,101 @@
...
+            f,
+            [
+                LhsAssignList::new(targets),
+                space(),
+                text("="),
+                space(),
+                value.format().with_options(Parenthesize::IfBreaks)
+            ]
+        )?;
+        Ok(())
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/statement/stmt_assign.rs:36]: NIt
```suggestion
        write!(
            f,
            [
                LhsAssignList::new(targets),
                space(),
                text("="),
                space(),
                value.format().with_options(Parenthesize::IfBreaks)
            ]
        )
```

File: crates/ruff_python_formatter/src/statement/stmt_assign.rs [hunk 3]
```diff
@@ -1,12 +1,101 @@
...
+    }
+}
+
+impl Format<PyFormatContext<'_>> for LeftRecursiveGrouping<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        // We shouldn't run into this case
+        let Some((current, tail)) = self.tail.split_last() else {
+            return Ok(())
+        };
+        // The last element can break if it's 84 chars on top level
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/statement/stmt_assign.rs:80]: What do you mean by 84 chars on top level? Can you add an example?

**Comment by konstin** [comment]
The solution was the `--delete-unreferenced-snapshots` in `cargo insta test --all --all-features --delete-unreferenced-snapshots`

**Pull request closed without merging**